src/entry.js

```diff
@@ -4,21 +4,22 @@ const Clock = require('./lamport-clock')
 const isDefined = require('./utils/is-defined')
 
 const IpfsNotDefinedError = () => new Error('Ipfs instance not defined')
+const encode = data => Buffer.from(JSON.stringify(data))
 
 class Entry {
   /**
    * Create an Entry
-   * @param {IPFS} ipfs - An IPFS instance
    * @param {string|Buffer|Object|Array} data - Data of the entry to be added. Can be any JSON.stringifyable data.
    * @param {Array<Entry|string>} [next=[]] Parents of the entry
    * @example
-   * const entry = await Entry.create(ipfs, 'hello')
+   * const entry = await Entry.createAndPublish(ipfs, 'hello')
    * console.log(entry)
-   * // { hash: "Qm...Foo", payload: "hello", next: [] }
+   * // { hash: null, payload: "hello", next: [] }
    * @returns {Promise<Entry>}
    */
-  static async create (ipfs, keystore, id, data, next = [], clock, signKey) {
+  static async createAndPublish (id, data, next = [], clock, nodeId, identity, ipfs) {
```

Comment:
I don't think we should rename this. We don't *publish* anything when the object gets added to IPFS, rather *persist* (but I don't think we need to exclusively add that to the function name).

Reply:
We shouldn't move `ipfs` as the last argument either as the general order of args throughout the code base is: 1) dependencies 2) required args 3) optional args.

As `identity` is a required argument, I would move it as the second parameter.

And perhaps to clarify the `id` arg, rename it to `logId`.

Then, the function signature would look like this:
```javascript
static async create (ipfs, identity, logId, data, next = [], clock)
```

Reply:
I believe we don't need `nodeId` anymore as `identity.publicKey` can be used for the clock ID (the public signing key was previously used as the clock id which is what we now encapsulate in `identity`).

Reply:
We do *store* the entry which maybe isnt expected with just "create" but I suppose passing in ipfs is a clue